Added support for Bitfunx memory card and Game ID functionality

MrPalloncini opened the pull request:
As stated in the title this will add the GameID functionality to automatically create and/or switch the memory card based on the game the xstation/uniboot loads. This can be enable/disable via config.h. It also adds support for the bitfunx memory card

MrPalloncini committed:
Add support for AliExpress Bitfunx card and GAMEID

Added support for Butfunx board as well as the GAMEID functionality, options can be enabled/disabled in config.h

MrPalloncini committed:
Update README.md

added some awesome contributors and general warnings for the new feature

MrPalloncini committed:
Update README.md

added another contributor i had to credit

MrPalloncini committed:
Fix a bunch of issues and PSIO support

Fix all of the previous issues except #1 and #5, also adds PSIO support and switching card with MemcardPro commands

dangiu commented:
Awesome work, I read in the README that it wasn't tested on PicoMemcard+. I'll try to test it next weekend to confirm that it works :)

MrPalloncini commented:
thanks, i'm fixing few things and adding new features in the meantime

MrPalloncini committed:
Fixes and improvements

Fixes many issues related to indexing, fix the ordering of the MCR files, add the option to use memcard pro commands to change the active memcard (still a bit buggy)

MrPalloncini commented:
This new release fixes a bunch of stuff and removes the ordering since is not needed. Has issues with Unirom tho

MrPalloncini committed:
Update README.md

added the new features and some thanks

MrPalloncini committed:
Fixed a shittons of issues

Thank you whitezombie2000 for every debug you did, also added the feature to switch memcard with the ode and with left and right, read the readme

MrPalloncini committed:
added the edited cmake file with the compile flag

i forgor

alex-free commented:
I just want to comment that I have implemented GameID support for the [Tonyhax International soft-mod](https://github.com/alex-free/tonyhax) and @MrPalloncini 's fork works great with my bitfunx psxmemcard.
